File: 26099/frontend/capture_screenshots.py
import asyncio
import os
from playwright.async_api import async_playwright

PORT = 5174
BASE_URL = f"http://localhost:{PORT}"
REVIEW_DIR = r"C:\Users\udayp\Documents\code\SIH26\26099\frontend\review"

PAGES = [
    ("/", "Dashboard"),
    ("/overview", "Overview"),
    ("/explorer", "Explorer"),
    ("/matching", "Matching"),
    ("/duplicates", "Duplicates"),
    ("/savings", "Savings"),
    ("/registry", "Registry"),
    ("/migration", "Migration"),
    ("/integration", "Integration"),
    ("/import", "Import"),
    ("/normalize", "Normalize"),
    ("/activity", "Activity"),
    ("/engine", "Engine"),
]

async def capture(p):
    browser = await p.chromium.launch(headless=True)
    page = await browser.new_page(viewport={"width": 1440, "height": 900})
    await page.goto(BASE_URL, wait_until="networkidle", timeout=30000)
    await page.wait_for_timeout(500)

    for path, label in PAGES:
        try:
            url = f"{BASE_URL}{path}"
            print(f"Capturing: {label} -> {url}")
            await page.goto(url, wait_until="networkidle", timeout=30000)
            await page.wait_for_timeout(500)

            safe = path.replace("/", "_").strip("_") or "home"
            screenshot_path = os.path.join(REVIEW_DIR, f"{safe}_1.png")
            await page.screenshot(path=screenshot_path, full_page=False)
            print(f"  Saved: {screenshot_path}")
        except Exception as e:
            print(f"  FAILED: {label} -> {e}")

    await browser.close()

async def main():
    os.makedirs(REVIEW_DIR, exist_ok=True)
    async with async_playwright() as p:
        await capture(p)

asyncio.run(main())
print("All screenshots captured!")

File: 26145/capture-screenshots.ts
import { chromium } from 'playwright';

const BASE = 'http://localhost:5178';
const OUT = 'C:/Users/udayp/Documents/code/SIH26/26145/review';

const PAGES = [
  { name: 'dashboard',      path: '/' },
  { name: 'live-threats',   path: '/live-threats' },
  { name: 'network-map',    path: '/network-map' },
  { name: 'ai-analyzer',    path: '/ai-analyzer' },
  { name: 'diode-lab',      path: '/diode-lab' },
  { name: 'analytics',      path: '/analytics' },
  { name: 'materials',      path: '/materials' },
  { name: 'activity',       path: '/activity' },
  { name: 'admin',          path: '/admin' },
  { name: 'integrations',   path: '/integrations' },
];

async function main() {
  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext({ viewport: { width: 1440, height: 900 } });

  for (const p of PAGES) {
    // ---- Dark mode ----
    const page = await context.newPage();
    await page.goto(`${BASE}${p.path}`, { waitUntil: 'networkidle', timeout: 20000 });
    await page.waitForTimeout(3000);
    await page.screenshot({ path: `${OUT}/dark/${p.name}.png`, fullPage: true });
    console.log(`  dark:  ${p.name}`);
    await page.close();

    // ---- Light mode ----
    const pageL = await context.newPage();
    await pageL.goto(`${BASE}${p.path}`, { waitUntil: 'networkidle', timeout: 20000 });
    await pageL.waitForTimeout(500);
    // Flip to light mode via theme toggle button
    await pageL.click('.theme-toggle');
    await pageL.waitForTimeout(1000);
    await pageL.screenshot({ path: `${OUT}/light/${p.name}.png`, fullPage: true });
    console.log(`  light: ${p.name}`);
    await pageL.close();
  }

  await browser.close();
  console.log('\nDone — screenshots saved to', OUT);
}

main().catch(e => { console.error(e); process.exit(1); });
